### ISSUE TITLE
data_source_okta_users omits admin_roles

### ISSUE THREAD

**Issue by exitcode0** [okta/terraform-provider-okta#1014]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v1.0.11
on darwin_arm64
+ provider registry.terraform.io/okta/okta v3.22.0
```
### Affected Resource(s)

- data_source_okta_users

### Terraform Configuration Files

```hcl
data "okta_users" "BLAH" {
  search {
    name = "profile.login"
    comparison = "eq"
    value = "exact.email.address@domain.com"
  }
}

output "testing" {
  value = data.okta_users.BLAH.users
}

output "testing2" {
  value = [for user in data.okta_users.BLAH.users : user.id if length(user.admin_roles) > 0]
}
```

### Expected Behavior

when the user in question is a super admin, i'd expect `testing2` output to contain the users ID and `testing1` to contain an entry similar to `"admin_roles" = toset(["SUPER_ADMIN"])`

### Actual Behavior

the output `testing2` outputs as empty and `testing1` correctly outputs the user object but the admins roles entry shows as `"admin_roles" = toset([])`

### Steps to Reproduce

1. `terraform refresh`

### References

- #0000

**Comment by monde** [comment]
Thanks @exitcode0 I'll look into this, thanks!

**Comment by exitcode0** [comment]
it seems that `data_source_okta_users` calls [/api/v1/users](https://github.com/okta/terraform-provider-okta/blob/master/okta/data_source_okta_users.go#L61)
which does not return admin roles for the users

the documentation for the datasource [Here](https://registry.terraform.io/providers/okta/okta/latest/docs/data-sources/users#admin_roles) would suggest that this is possible

So I guess either the doco here is missleading and should be updated or the implementation or `data_source_okta_users` needs to be changed to include this as an option 
perhaps with a `skip_roles` similar to `data_source_okta_user` with a default value of true

**Comment by exitcode0** [comment]
My workaround for this for now in case someone else comes across this issue

```hcl
resource "okta_resource_set" "okta_admins" {
  label       = "okta_admins"
  description = "okta_admins"

  resources = [
    "https://${local.org_name}.${local.base_url}/api/v1/groups/${okta_group.okta_admins.id}/users",
  ]
}
resource "okta_group" "okta_admins" {
  name        = "okta_admins"
  description = "Users with okta admin permissions. Membership to this group is managed by terraform."
  skip_users  = true

}
resource "okta_group_memberships" "okta_admins" {
  group_id = okta_group.okta_admins.id
  users = [
    for user in data.okta_app_user_assignments.okta_admin_console.users :
    user
  ]
}
data "okta_app" "okta_admin_console" {
  label = "Okta Admin Console"
}
data "okta_app_user_assignments" "okta_admin_console" {
  id = data.okta_app.okta_admin_console.id
}
```

**Comment by monde** [comment]
being addressed in #1159
